iati/resources.py

```diff
@@ -57,88 +61,89 @@
 """The name of a file containing an Organisation Schema."""
 
 
-def get_codelist_paths(version=iati.version.STANDARD_VERSION_ANY):
+@iati.version.decimalise_integer
+@iati.version.allow_possible_version
+def get_codelist_paths(version):
     """Find the paths for all Codelists at the specified version of the Standard.
 
     Args:
-        version (str): The version of the Standard to return the Codelists for. Defaults to iati.version.STANDARD_VERSION_ANY.
+        version (str / int / Decimal / iati.Version): The version of the Standard to return the Codelists for.
+            Decimal: Return paths for the specified version of the Standard.
+            Integer: Return paths for the latest Decimal version within the given integer.
+            Version-independent: Return paths for Non-Embedded Codelists.
 
     Raises:
-        ValueError: When a specified version is not a valid version of the IATI Standard.
+        TypeError: When a specified version is of a type that cannot represent an IATI version number.
+        ValueError: When a specified version is of the correct type, but cannot represent a version of the IATI Standard.
 
     Returns:
         list(str): A list of paths to all of the Codelists at the specified version of the Standard.
 
     Todo:
-        Further exploration needs to be undertaken in how to handle pre-1.04 versions of the Standard.
+        Look to provide an argument that allows the returned list to be restricted to only Embedded or only Non-Embedded Codelists.
```

Comment:
parameter that is `embedded=True` or `non_embedded=True`???

Reply:
Adding functionality along these lines is out of scope for this PR so will need to be designed and undertaken separately.

Reply:
https://github.com/IATI/pyIATI/pull/297/commits/b26c2c61e475b6c0187e4f390064a204516836eb